tools/code_analyzer/op_deps_processor.py

```diff
@@ -0,0 +1,109 @@
+"""
+This util is used to parse op_deps_pass output (in yaml) and convert it into
+other formats for downstream use cases. It is not used by OSS cmake build.
+
+To run this file by hand from the root of the PyTorch repository, run:
+
+python -m tools.code_analyzer.op_deps_processor \
+  --op-dependency build_code_analyzer/work/torch_result.yaml \
+  --output pt_deps.bzl
+"""
+
+import argparse
+import yaml
+
+from ..autograd.utils import CodeTemplate
+
+BAZEL_OUTPUT = CodeTemplate("""\
+# Generated for selective build without using static dispatch.
+# Manually run the script to update:
+# ANALYZE_TORCH=1 DEPLOY=1 tools/code_analyzer/build.sh
+TORCH_DEPS = {
+${ops}
+}
+""")
+
+BAZEL_OP = CodeTemplate("""\
+    "${op_name}": [
+${op_deps}
+    ],
+""")
+
+BAZEL_OP_DEP = CodeTemplate("""\
+        "${dep_name}",
+""")
+
+DOT_OUTPUT = CodeTemplate("""\
+digraph {
+    layout="circo";
+${ops}
+}
+""")
+
+DOT_OP = CodeTemplate("""\
+${op_deps}
+""")
+
+DOT_OP_DEP = CodeTemplate("""\
+    "${op_name}" -> "${dep_name}";
+""")
+
+
+def load_op_deps(fname):
+    with open(fname, 'r') as stream:
+        return yaml.safe_load(stream)
+
+
+def convert(fname, graph, OUTPUT, OP, OP_DEP):
```

Comment:
nit: might be slightly easier to distinguish between these params and the template globals if the params were lowercase

Reply:
hmm, on the other hand I want to distinguish between code template v.s. code after substitution; maybe I can use "_template" suffix instead...

Reply:
ooo yeah, `_template` is perfect